guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/CacheSerializer.java

```diff
@@ -46,10 +105,26 @@ public void serializeWithType(Cache<?, ?> value, JsonGenerator gen, SerializerPr
         _writeContents(value, gen, ctxt);
         typeSer.writeTypeSuffix(gen, typeIdDef);
     }
+    
+    /*
+    /**********************************************************
+    /* Internal helper methods
+    /**********************************************************
+     */
 
-    // Just a stub in case we have time to implement proper (if optional) serialization
-    protected void _writeContents(Cache<?, ?> value, JsonGenerator g, SerializerProvider ctxt)
-        throws JacksonException
+    protected void _writeContents(Cache<?, ?> cache, JsonGenerator gen, SerializerProvider provider)
+        throws IOException
     {
-    }    
+        final Set<String> ignored = _ignoredEntries;
+        for (Map.Entry<?, ?> entry : cache.asMap().entrySet()) {
+            Object key = entry.getKey();
+            Object value = entry.getValue();
+            // ignore? 
+            if (ignored != null && ignored.contains(key.toString())) {
+                continue;
+            }
+            gen.writeFieldName(key.toString());
+            provider.defaultSerializeValue(value, gen);
```

Comment:
This won't support polymorphism and is not as efficient as if serializer was fetched eagerly (like regular `Map`s will do, if possible; or if not possible, retain (possibly generic) element `JavaType`) -- but for a minimal implementation it's a clear improvement so that's fine.